src/Controller/UserController.php

```diff
@@ -135,4 +136,12 @@ public function confirmInvitationAction(Request $request, string $code,
             'form' => $form->createView()
         ]);
     }
+
+    /**
+     * @Route("/mode-pro", name="pro_account_cart")
+     */
+    public function proCartAction(Request $request)
```

Comment:
Instead of adding a new action, you should use the mechanism of Sylius for resolving/storing the channel. 
See [how it's configured in DI](https://github.com/Sylius/Sylius/blob/8e50539ce5015d98403b6c25637e4b6bb1b6ddf9/src/Sylius/Bundle/ChannelBundle/Resources/config/services.xml#L40-L60), and see [HostnameBasedRequestResolver](https://github.com/Sylius/Sylius/blob/8e50539ce5015d98403b6c25637e4b6bb1b6ddf9/src/Sylius/Component/Channel/Context/RequestBased/HostnameBasedRequestResolver.php) for an example of resolving the channel based on the request. 

Just look at how [FakeChannelContext](https://github.com/Sylius/Sylius/blob/8e50539ce5015d98403b6c25637e4b6bb1b6ddf9/src/Sylius/Bundle/ChannelBundle/Context/FakeChannel/FakeChannelContext.php) works (also, see the [service definition](https://github.com/Sylius/Sylius/blob/8e50539ce5015d98403b6c25637e4b6bb1b6ddf9/src/Sylius/Bundle/ChannelBundle/Resources/config/services/integrations/debug.xml#L26-L31)). It should be implemented in the same way. 

---

Actually, we already have [a service tagged with `sylius.context.channel.request_based.resolver`](https://github.com/coopcycle/coopcycle-web/blob/458c4549c9dc3c860177bc0107df4946851286ef/app/config/services.yml#L622-L624), which resolves the channel from a HTTP header.